service/controller/control_plane.go

```diff
@@ -67,14 +88,59 @@ func NewControlPlane(config ControlPlaneConfig) (*ControlPlane, error) {
 func newControlPlaneResourceSets(config ControlPlaneConfig) ([]*controller.ResourceSet, error) {
 	var err error
 
-	var resourceSet *controller.ResourceSet
+	var certsSearcher *certs.Searcher
 	{
-		c := controlPlaneResourceSetConfig{
-			G8sClient: config.K8sClient.G8sClient(),
+		c := certs.Config{
 			K8sClient: config.K8sClient.K8sClient(),
 			Logger:    config.Logger,
+		}
 
-			HostAWSConfig: config.HostAWSConfig,
+		certsSearcher, err = certs.NewSearcher(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var randomKeysSearcher randomkeys.Interface
+	{
+		c := randomkeys.Config{
+			K8sClient: config.K8sClient.K8sClient(),
+			Logger:    config.Logger,
+		}
+
+		randomKeysSearcher, err = randomkeys.NewSearcher(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSet *controller.ResourceSet
+	{
+		c := controlPlaneResourceSetConfig{
+			G8sClient:          config.K8sClient.G8sClient(),
+			K8sClient:          config.K8sClient.K8sClient(),
+			Logger:             config.Logger,
+			CertsSearcher:      certsSearcher,
+			RandomKeysSearcher: randomKeysSearcher,
+
+			CalicoCIDR:                config.CalicoCIDR,
+			CalicoMTU:                 config.CalicoMTU,
+			CalicoSubnet:              config.CalicoSubnet,
+			ClusterIPRange:            config.ClusterIPRange,
+			DockerDaemonCIDR:          config.DockerDaemonCIDR,
+			EncrypterBackend:          config.EncrypterBackend,
+			IgnitionPath:              config.IgnitionPath,
+			ImagePullProgressDeadline: config.ImagePullProgressDeadline,
+			InstallationName:          config.InstallationName,
+			NetworkSetupDockerImage:   config.NetworkSetupDockerImage,
+			PodInfraContainerImage:    config.PodInfraContainerImage,
+			RegistryDomain:            config.RegistryDomain,
+			SSHUserList:               config.SSHUserList,
+			SSOPublicKey:              config.SSOPublicKey,
+			VaultAddress:              config.VaultAddress,
+
+			HostAWSConfig:  config.HostAWSConfig,
+			Route53Enabled: config.Route53Enabled,
```

Comment:
So far the blocks were separated into dependency and setting blocks. What order did you want to follow here?